profiler/profiler.go

```diff
@@ -154,6 +158,15 @@ func (p *profiler) run() {
 	if _, ok := p.cfg.types[BlockProfile]; ok {
 		runtime.SetBlockProfileRate(p.cfg.blockRate)
 	}
+	if s, ok := p.cfg.statsd.(statsdGauge); ok {
+		// If the statsd client supports gauges (e.g.
+		// github.com/datadog/datadog-go/v5/statsd) then we can send
+		// gauges reporting the current profiler configuration
+		s.Gauge("datadog.profiler.go.profile_period", p.cfg.period.Seconds(), nil, 1)
+		s.Gauge("datadog.profiler.go.cpu_duration", p.cfg.cpuDuration.Seconds(), nil, 1)
+		s.Gauge("datadog.profiler.go.block_profile_rate", float64(p.cfg.blockRate), nil, 1)
+		s.Gauge("datadog.profiler.go.mutex_profile_fraction", float64(p.cfg.mutexFraction), nil, 1)
```

Comment:
NIT: Using `runtime.SetMutexProfileFraction(-1)` to get the mutex profile fraction should be more reliable (some apps might call `runtime.SetMutexProfileFraction()` rather than going through our config options. Unfortunately the block profile API doesn't offer this.